packages/ui-kit/lib/components/video-player.tsx

```diff
@@ -0,0 +1,164 @@
+    const timeoutRef = useRef<any>(null);
+    
+    // Check if videoId exists at all
+    useEffect(() => {
+        if (!videoId || videoId.trim() === '') {
+            setVideoError(true);
+            setIsLoading(false);
+            onErrorCallback(dictionary.components.videoPlayer.videoErrorText, new Error('Missing video ID'));
+        } else {
+            setVideoError(false);
+            // Only show player if there's a valid ID and no thumbnail, or if user clicked thumbnail
+            setShowPlayer(!thumbnailUrl);
+            
+            // Reset loading state when ID changes
+            setIsLoading(true);
+        }
+        
+        // Clear any existing timeout when ID changes
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+        
+        // Set timeout to show error if video doesn't load within the specified time
+        // This helps prevent prolonged flickering
+        if (videoId) {
+            timeoutRef.current = setTimeout(() => {
+                if (isLoading) {
+                    setVideoError(true);
+                    setIsLoading(false);
+                    onErrorCallback(dictionary.components.videoPlayer.videoErrorText, new Error('Video load timeout'));
+                }
+            }, loadTimer);
+        }
+        
+        return () => {
+            if (timeoutRef.current) {
+                clearTimeout(timeoutRef.current);
+            }
+        };
+    }, [videoId, thumbnailUrl, dictionary, onErrorCallback, loadTimer]);
+
+    const handleThumbnailClick = () => {
+        setShowPlayer(true);
+        setAutoPlay(true);
+    };
+
+    const handleVideoError = (event: any) => {
+        setVideoError(true);
+        setIsLoading(false);
+        onErrorCallback(dictionary.components.videoPlayer.videoErrorText, event);
+        
+        // Clear timeout since we got a definitive error
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+    };
+    
+    const handlePlayerLoaded = () => {
+        setIsLoading(false);
+        
+        // Clear timeout since player loaded successfully
+        if (timeoutRef.current) {
+            clearTimeout(timeoutRef.current);
+        }
+    };
+
+    // Error component
+    const ErrorDisplay = () => (
+        <div className="rounded-medium w-full min-w-[18rem] h-[16rem] bg-base-neutral-700 flex items-center justify-center p-4">
+            <span className="text-text-secondary text-md">
+                {dictionary.components.videoPlayer.videoErrorText}
+            </span>
+        </div>
+    );
+
+    // Loading component - shown while waiting for player to initialize
+    const LoadingDisplay = () => (
+      <div className='h-full w-full items-center justify-center flex'>
+        <div className="w-6 h-6 border-4 border-base-neutral-700 border-t-transparent rounded-full animate-spin"></div>
+        </div>
+    );
```

Comment:
Moving these nested components out of the main one will prevent their recreation on each render